### PR TITLE
Focus style in styleguide

### PR DESCRIPTION
**Pull request by katarzynatobis**
Focus style for style guide:
- 2 types of focus:
    - `focusStyle` - default one
    - `focusTextStyle` - should be used for inline, wrapped elements
- new reusable mixins: 
    - `applyFocusStyle` & `applyFocusTextStyle` - only with focus style
    - `applyFocus` & `applyFocusText` - provide logic when focus should be visible

**focusStyle**:
Screen Recording 2022-11-02 at 12 06 32 (1)

**focusTextStyle**:
199474689-d0fc8936-7785-48d6-b6e1-ea0cbd2db295

Components using `focusStyle`: Accordion, Avatar, Button, Dialog, FileHandler, HomeButton, IconAsButton, Label, Rating, Search, SkipLink, native interactive elements besides `<a>`
Components using `focusTextStyle`: Link, Checkbox, Radio, Select, Dropdown, Search, Textarea, native `<a>`

Focus colors can be changed by overwriting css variables: `--focusColorOuter`, `--focusColor`, `--focusColorInner`

Accordion `focusStyle`
https://user-images.githubusercontent.com/60467496/199485757-37643eac-8e78-48c1-b342-555b05ce7d4a.mov

Avatar `focusStyle`
https://user-images.githubusercontent.com/60467496/199485873-825660eb-5860-44df-96fe-329de380c451.mov

Button `focusStyle`
https://user-images.githubusercontent.com/60467496/199485816-bdfe309a-d958-4006-92cf-4cb9d22ce7d3.mov

Dropdown `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199485918-0d454745-a1ec-4908-b450-5103ebbb8d91.mov

FileHandler `focusStyle`
https://user-images.githubusercontent.com/60467496/199486021-4d82c55a-1adc-4076-a5ff-712bd2ff34fb.mov

Checkbox `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486096-289928fe-4f56-456b-99b6-ab3068a321ed.mov

Input `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486173-41f29d85-d8f5-491b-befc-d9b8c45c3d24.mov

Radio `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486224-481cd909-7e45-47fa-8a36-5b8ba001f651.mov

Select `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486296-60e7c979-1747-46c5-9e89-9c09530b1501.mov

Textarea `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486353-b8236294-2e76-43ed-9189-912c399f1a0b.mov

HomeButton `focusStyle`
https://user-images.githubusercontent.com/60467496/199486434-4dc33600-1891-469d-9ec2-71229a1c99bb.mov

IconAsButton `focusStyle`
https://user-images.githubusercontent.com/60467496/199486466-7ad90125-85d6-4055-b21d-5ddd476c04ef.mov

Label `focusStyle`
https://user-images.githubusercontent.com/60467496/199486506-cc4a5d67-7c11-4031-9fd8-1a7072586982.mov

Radio `focusStyle`
https://user-images.githubusercontent.com/60467496/199486557-821f45af-bf4a-4450-8da7-51c359015587.mov

Search `focusStyle` & `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486595-e28ce8b7-b818-4c03-9e88-5526ac5d4e27.mov

SkipLink `focusStyle`
https://user-images.githubusercontent.com/60467496/199486668-9798dc15-5702-4266-a97a-e8e666df342d.mov

Link `focusTextStyle`
https://user-images.githubusercontent.com/60467496/199486842-b77f03a5-d017-4dc5-89a8-88da829f4a26.mov

Dialog `focusStyle`
https://user-images.githubusercontent.com/60467496/199486795-3814700a-3434-4f33-a1c9-670d49a4603b.mov

### CODE REVIEW

File: src/sass/_utils.scss [hunk 1]
```diff
@@ -84,7 +84,7 @@
 
 .sg-visually-hidden {
   position: absolute;
-  width: 1px;
+  width: 0px;
```
Comment (coderitual) [inline src/sass/_utils.scss:87]: could you please explain?

**Comment by coderitual** [comment]
thx @katarzynatobis , please add description to the PR so it's easier to review.

File: src/sass/_utils.scss (hunk 1 above)
Comment (katarzynatobis) [reply src/sass/_utils.scss]: It was supposed to remove this tiny little "pimple" from Link focus area:
Screenshot 2022-11-03 at 12 04 27
But I figured out how to fix it in a different way.

File: src/sass/_focus.scss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+@mixin applyFocusTextStyle() {
+  outline-color: var(--focusColor);
+  outline-width: 2px;
+  outline-offset: 4px;
+  outline-style: auto;
+}
+
+@mixin applyFocus() {
+  transition: box-shadow $durationQuick1 $easingExit;
```
Comment (coderitual) [inline src/sass/_focus.scss:57]: why we need this transition + animation?

**Comment by coderitual** [comment]
@katarzynatobis border radius in checkbox seems incorrect, also it is not using custom value
CleanShot 2022-11-03 at 12 29 24
CleanShot 2022-11-03 at 12 29 18

**Comment by coderitual** [comment]
Probably need to be slightly adjusted
CleanShot 2022-11-03 at 16 54 06

File: src/components/skip-link/SkipLink.stories.mdx [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
 
 export const play = async ({canvasElement}) => {
   const canvas = within(canvasElement);
   const link = canvas.getByRole('link', {
     name: /skip to main content/i,
   });
-  link.focus();
+  await sleep(300);
```
Comment (coderitual) [inline src/components/skip-link/SkipLink.stories.mdx:15]: @katarzynatobis i think you can use `  chromatic: { delay: 300 },` in chromatic story instead

Comment (katarzynatobis) [reply src/components/skip-link/SkipLink.stories.mdx]: I tried this custom function and chromatic delay - both ways did not work (for some reason css animation is not played). Finally, I just removed from snapshot the animation of focus ring.

File: src/sass/_focus.scss (hunk 2 above)
Comment (coderitual) [reply src/sass/_focus.scss:57]: @katarzynatobis not sure if you saw this comment

File: src/components/skip-link/SkipLink.stories.mdx (hunk 3 above)
Comment (coderitual) [reply src/components/skip-link/SkipLink.stories.mdx]: cool let's use that

File: src/sass/_focus.scss (hunk 2 above)
Comment (katarzynatobis) [reply src/sass/_focus.scss:57]: Entry and exit animation differ significantly. The exit one is much simpler:
Screenshot 2022-11-04 at 16 49 47
and this is why I used `transition` to animate. 
While the entry one is more difficult - each color has its own delay, easing and time, but it is still the same `box-shadow` property.

**Review by coderitual — APPROVED** [review]

**Pull request merged**